[CCR] Add auto follow pattern detail panel

jen-huang opened the pull request:
This PR adds:

- Plugin dependency declarations
- Handling of partial failures for removing clusters
- Auto follow pattern detail panel, which is deep-linked with `pattern` query param

![image](https://user-images.githubusercontent.com/1965714/50060817-c2927b00-014d-11e9-9ce4-6ac896b478cb.png)


To do:
- [ ] Add link in detail panel to matching indices in index management (as a table filter)

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/4029/)

elasticmachine commented:
Pinging @elastic/es-ui

yaronp68 commented:
@jen-huang great job. Two minor comments:

1. With previous fly-outs we presented the value next to the field name on the same row (see index management)
2. Leader patterns should IMO be either present as a spaces separated tags or as a bulleted list so it is easy to read.

sebelga commented on x-pack/plugins/remote_clusters/public/store/actions/remove_clusters.js:
```diff
@@ -27,33 +25,46 @@ export const removeClusters = (names) => async (dispatch, getState) => {
+      .catch(() => removalErrors.push(name));
+  });
+
+  await Promise.all([
```
This would be better if done server side like I did with auto-follow pattern. We can maybe leave it as an optimization fix for later though.

cjcenizal commented on x-pack/plugins/cross_cluster_replication/public/app/sections/home/auto_follow_pattern_list/components/detail_panel/detail_panel.js:
```diff
@@ -0,0 +1,303 @@
+
+            <EuiSpacer size="s" />
+
+            <EuiFlexGroup>
```
I just found a visual bug. When there is no `prefix` or `suffix` defined, could we render `No suffix` and `No prefix` inside of `<em>` tags? Right now they appear blank:

![image](https://user-images.githubusercontent.com/1238659/50114255-f5de1400-01f8-11e9-968b-45c0efb80db8.png)

jen-huang commented on x-pack/plugins/cross_cluster_replication/public/app/sections/home/auto_follow_pattern_list/components/detail_panel/detail_panel.container.js:
```diff
@@ -0,0 +1,51 @@
+  return {
+    isDetailPanelOpen: isDetailPanelOpen(state),
+    autoFollowPattern: getDetailPanelAutoFollowPattern(state),
+    autoFollowPatternName: getDetailPanelAutoFollowPatternName(state),
```
the deep link could also refer to a nonexistent pattern, which will result in a null `autoFollowPattern` object

jen-huang commented:
Thank you for the reviews, @cjcenizal and @sebelga. I've addressed most of the feedback, including the most important URL encoding of the pattern name param.

As far as using `selectedId` for both editing and viewing details - I found the mental model unnecessarily complex since the meaning of that ID changes based on another part of the state (a flag called `isDetailPanelOpen` for example). I opted to keep `detailPanelId` (no changes) to be explicit about the meaning, but we can definitely revisit this area later. 😄

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/4257/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/4260/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/4266/)

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/4271/)

sebelga commented:
Ok, agree with cj that time is too tight to risk FF 😊 But to me, the mental model is more complex with the current implementation (more action types, more reducers, more selectors, more code to read, more tests to write....). And for the future... Yagni ! (http://wiki.c2.com/?YouArentGonnaNeedIt) 😄
